corehq/apps/reports_core/filters.py

```diff
@@ -139,7 +139,9 @@ def value(self, **kwargs):
 
         def date_or_nothing(param):
             if param:
-                if self.compare_as_string:
+                if isinstance(param, date):
+                    return datetime.combine(param, time())
```

Comment:
This is an artifact of passing in filters directly from the saved report configuration rather than url parameters.

Reply:
@czue I got very distracted but would like to wrap this up. After picking this back up, I realized this is not a very elegant solution. Would you agree that instead of adding an additional case to the DatespanFilter to accept actual datetime objects, the optimal solution would be to return a properly formatted dictionary from saved_report_config here:

https://github.com/dimagi/commcare-hq/blob/2ddc6e48a27353c1bf7e0c002cec5801e5f1e840/corehq/apps/userreports/reports/view.py#L256-L261

Reply:
Ah I see. Just to confirm my understanding:

1. Prior to this change it always returned a `request_dict` that had serialized data
2. The code that was modified in this PR changes it to return a native (couch?) dict that has python objects in it.
3. You're proposing changing the new code to serialize the filters before being returned, so it looks like a `request_dict`?

That makes sense to me. Wondering if there's a better name for "get me the serialized version of the filters" than `filter_context` but changing the name is probably out of scope unless easy.